model-optimizer/extensions/middle/StridedSliceNormalizer.py

```diff
@@ -0,0 +1,202 @@
+        return [LayoutChangeForConstantShapePaths]
+
+    def find_and_replace_pattern(self, graph: Graph):
+        ss_nodes = graph.get_op_nodes(op='StridedSlice')
+        for node in ss_nodes:
+            self.normalize_strided_slice(graph, node)
+
+            PermuteAttrs.create_permute_attrs(node, attrs=[('shrink_axis_mask', 'input:0'),
+                                                           ('new_axis_mask', 'input:0'),
+                                                           ('ellipsis_mask', 'input:0'),
+                                                           ('begin_mask', 'input:0'),
+                                                           ('end_mask', 'input:0')])
+
+            if len(node.in_port(0).data.get_shape()) < 4 and len(node.out_port(0).data.get_shape()) > 3:
+                # for the cases when we insert new_axis
+                PermuteInputs().set_input_permutation(node.in_node(1), node, 'output:0', 'shape')
+                PermuteInputs().set_input_permutation(node.in_node(2), node, 'output:0', 'shape')
+                PermuteInputs().set_input_permutation(node.in_node(3), node, 'output:0', 'shape')
+            else:
+                PermuteInputs().set_input_permutation(node.in_node(1), node, 'input:0', 'shape')
+                PermuteInputs().set_input_permutation(node.in_node(2), node, 'input:0', 'shape')
+                PermuteInputs().set_input_permutation(node.in_node(3), node, 'input:0', 'shape')
+
+    def normalize_strided_slice(self, graph: Graph, node: Node):
+        input_shape = node.in_port(0).data.get_shape()
+        input_rank = len(input_shape)
+        slice_rank = node.in_port(1).data.get_shape()[0]
+
+        slice_mask_names = ['begin_mask', 'end_mask', 'new_axis_mask', 'shrink_axis_mask', 'ellipsis_mask']
+        # allign masks sizes with slice_rank (not confuse with extending mask_alligment != mask_extending)
+        for mask_name in slice_mask_names:
+            num = slice_rank - len(node[mask_name])
+            val = 0 if mask_name not in ['begin_mask', 'end_mask'] else 1  # extend with ones only for begin and end
+            node[mask_name] = np.append(node[mask_name], [val] * num).astype(int)
+
+        if np.any(node.ellipsis_mask):
+            idx = np.nonzero(node.ellipsis_mask)
+            assert len(idx[0]) == 1, 'only one ellipsis_mask nonzero value is allowed'
+            ellipsis_start = idx[0][0]
+            # since we don't use begin, end values
+            node.begin_mask[ellipsis_start] = 0
+            node.end_mask[ellipsis_start] = 0
+
+            num = input_rank - slice_rank + np.count_nonzero(node.new_axis_mask[ellipsis_start:])
+
+            # unroll ellipsis for masks
+            node.ellipsis_mask[ellipsis_start] = 0
+            for mask_name in slice_mask_names:
+                node[mask_name] = np.insert(node[mask_name], ellipsis_start + 1, [0] * num).astype(int)
```

Comment:
We have exactly the same logic in the `infer` function. it should be removed from one place or if it is really needed in two places then implementation should be moved to a common function. However, I expect that it should be used in one place only (infer function)

Reply:
It's not exactly the same. In shape_infer we were not changing node attributes as we do here and we change begin, end values in shape_infer in the same way as masks but here begins and end are untouchable we insert nodes instead of values and some other nuances that complicate generalization.

But anyway, now I have rewritten shape_infer and these lines are present only here. Indeed, they are more important here than in shape_infer.